python/semantic_kernel/kernel.py

```diff
@@ -383,6 +383,29 @@ async def invoke_function_call(
             chat_history.add_message(message=frc.to_chat_message_content())
             return None
 
+        # Check for missing or unexpected parameters
+        required_param_names = {param.name for param in function_to_call.parameters if param.is_required}
+        received_param_names = set(parsed_args or {})
+
+        missing_params = required_param_names - received_param_names
+        unexpected_params = received_param_names - {param.name for param in function_to_call.parameters}
+
+        if missing_params or unexpected_params:
+            msg_parts = []
+            if missing_params:
+                msg_parts.append(f"Missing required argument(s): {sorted(missing_params)}.")
+            if unexpected_params:
+                msg_parts.append(f"Received unexpected argument(s): {sorted(unexpected_params)}.")
+            msg = " ".join(msg_parts) + " Please revise the arguments to match the function signature."
```

Comment:
this is not a good idea, the KernelArguments used might be propagated onwards towards this, and then this will raise, I think the behavior should be:

- missing_params: this
- unexpected_params: return this msg only if the function call fails, in addition to the actual error

Reply:
I re-positioned the method to be before we add in the kernel args. Good catch. Also added a unit test.